hail/src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -2465,88 +2465,91 @@ case class MatrixExplodeCols(child: MatrixIR, path: IndexedSeq[String]) extends
   }
 }
 
-/**
- * This is inteded to be an inverse to LocalizeEntries on a MatrixTable
- *
- * Some notes on semantics,
- * `rowsEntries`'s globals populate the resulting matrixtable, `cols`' are discarded
- * `entryFieldName` must refer to an array of structs field in `rowsEntries`
- * all elements in the array field that `entriesFieldName` refers to must be present. Furthermore,
- * all array elements must be the same length, though individual array items may be missing.
- */
-case class UnlocalizeEntries(rowsEntries: TableIR, cols: TableIR, entryFieldName: String) extends MatrixIR {
-  private val m = Map(entryFieldName -> MatrixType.entriesIdentifier)
-  private val entryFieldIdx = rowsEntries.typ.rowType.fieldIdx(entryFieldName)
-  private val entryFieldType = rowsEntries.typ.rowType.types(entryFieldIdx)
-  private val newRowType = rowsEntries.typ.rowType.rename(m)
-
-  entryFieldType match {
-    case TArray(TStruct(_, _), _) => {}
-    case _ => fatal(s"expected entry field to be an array of structs, found ${ entryFieldType }")
+/** Create a MatrixTable from a Table, where the column values are stored in a
+  * global field 'colsFieldName', and the entry values are stored in a row
+  * field 'entriesFieldName'.
+  */
+case class CastTableToMatrix(
+  child: TableIR,
+  entriesFieldName: String,
+  colsFieldName: String,
+  colKey: IndexedSeq[String]
+) extends MatrixIR {
+
+  private val m = Map(entriesFieldName -> MatrixType.entriesIdentifier)
+
+  private val entriesFieldType = child.typ.rowType.fieldType(entriesFieldName)
+
+  private val Field(_,
+    TArray(colType@TStruct(_, _), _),
+    colsFieldIdx
+  ) = child.typ.globalType.field(colsFieldName)
+
+  private val newRowType = child.typ.rowType.rename(m)
+
+  entriesFieldType match {
```

Comment:
It'd be nice if both this (entriesFieldType) and the colsFieldType were checked in the same way (maybe throw an error for the colsFieldName extraction too instead of just a match error?).